internal/app.go

```diff
@@ -22,51 +31,115 @@ const (
 )
 
 type App struct {
-	commit  string
-	version string
+	grpcConn *grpc.GrpcConnection
+	commit   string
+	version  string
 }
 
 func NewApp(commit, version string) *App {
-	return &App{commit, version}
+	return &App{nil, version, commit}
 }
 
-func (a *App) Run(ctx context.Context) error {
+func (a *App) Run(ctx context.Context) (err error) {
+	a.handleSignal(ctx)
+
 	config.Init(a.version, a.commit)
 
 	config.RegisterRunner(func(_ *cobra.Command, _ []string) {
-		err := config.RegisterConfigFile()
+		err = config.RegisterConfigFile()
 		if err != nil {
 			slog.ErrorContext(ctx, "Failed to load configuration file", "error", err)
 			return
 		}
 
-		agentConfig, err := config.ResolveConfig()
-		if err != nil {
-			slog.ErrorContext(ctx, "Invalid config", "error", err)
+		agentConfig, configErr := config.ResolveConfig()
+		if configErr != nil {
+			slog.ErrorContext(ctx, "Invalid config", "error", configErr)
 			return
```

Comment:
Now that we've renamed the error to `configErr`, when we hit `return` here `err` is nil. Are we sure we want to return a nil `err` here?

Reply:
it is inside config.RegisterRunner and the return breaks out of the function. I'm not sure what the best way to do this is since the Run function can't get that error because previously, the ```err := config.Execute(ctx)``` overrode the err value

Reply:
Ah right, missed that. Dunno could we change the `RegisterRunner` function so that we pass it a lambda that return an `error` instead of the what it's like now? So the new implementation would be something like:
```go
func RegisterRunner(r func(cmd *cobra.Command, args []string) error) {
	RootCommand.RunE = r
}
```

Reply:
ah nice, will check this